packages/frontend/amp/components/Blocks.tsx

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import { Elements } from '@frontend/amp/components/Elements';
+import { css } from 'emotion';
+import { pillarPalette } from '@frontend/lib/pillars';
+import { palette } from '@guardian/pasteup/palette';
+import { blockLink } from '@frontend/amp/lib/block-link';
+import { findBlockAdSlots } from '@frontend/amp/lib/find-adslots';
+import { WithAds } from '@frontend/amp/components/WithAds';
+
+const blockStyle = (pillar: Pillar) => css`
+    padding: 6px 10px 12px;
+    background-color: ${palette.neutral[100]};
+    border-top: 1px solid ${pillarPalette[pillar].dark};
+    border-bottom: 1px solid ${palette.neutral[93]};
+    margin-bottom: 12px;
+`;
+
+// TODO ad handling (currently done in elements, which is wrong, so let's lift
+// that out and have an Ad element type we match against
+export const Blocks: React.SFC<{
+    blocks: Block[];
+    pillar: Pillar;
+    edition: Edition;
+    section?: string;
+    contentType: string;
+    switches: Switches;
+    commercialProperties: CommercialProperties;
+    url: string;
+    shouldHideAds: boolean;
+}> = ({
+    blocks,
+    pillar,
+    edition,
+    section,
+    contentType,
+    switches,
+    commercialProperties,
+    url,
+    shouldHideAds,
+}) => {
+    // TODO add last updated for blocks to show here
+    const liveBlogBlocks = blocks.map(block => {
+        return (
+            <div
+                id={block.id}
+                data-sort-time={block.createdOn}
+                key={block.id}
+                className={blockStyle(pillar)}
+            >
+                {block.createdOnDisplay && (
+                    <a href={blockLink(url, block.id)}>
+                        {block.createdOnDisplay}
+                    </a>
+                )}
+                {block.title && <h2>{block.title}</h2>}
+                <Elements
+                    pillar={pillar}
+                    elements={block.elements}
+                    // stuff for ads
+                    edition={edition}
+                    section={section}
+                    contentType={contentType}
+                    switches={switches}
+                    commercialProperties={commercialProperties}
+                    isImmersive={false}
+                    shouldHideAds={true} // always no ads *within* blocks for liveblogs
+                />
+                {block.lastUpdatedDisplay && (
+                    <span>{block.lastUpdatedDisplay}</span>
+                )}
+            </div>
+        );
+    });
+
+    if (shouldHideAds) {
+        return <>liveBlogBlocks</>;
```

Comment:
Not tested locally, but is this returning JSX anyway, so might not need the warpper?

```suggestion
        return liveBlogBlocks;
```

Reply:
this is a bug in fact, which I've just fixed. really we need `<>{liveBlogBlocks}</>`